programs/vote/src/vote_state/mod.rs

```diff
@@ -133,7 +133,9 @@ impl From<VoteStateUpdate> for VoteTransaction {
 
 // utility function, used by Stakes, tests
 pub fn from<T: ReadableAccount>(account: &T) -> Option<VoteState> {
-    VoteState::deserialize(account.data()).ok()
+    bincode::deserialize::<VoteStateVersions>(account.data())
+        .map(|versioned| versioned.convert_to_current())
+        .ok()
 }
```

Comment:
this function is underlies `stake_state::create_account`, which is used in the core runtime. we retain the old behavior for maximum safety. there are inputs the new parser will reject which bincode would have accepted, although it is unlikely any such vote accounts exist. this will be reverted when we feature gate upgrade the vote program to the new parser